### DISCUSSION TITLE
Detectron2 - Same Code / Different Plattforms / highly divergent results

### DISCUSSION THREAD

**Discussion by Natriumpikant** [facebookresearch/detectron2#4365] [General]
I use different hardware to benchmark multiple possibilites. The Code runs in a jupyter Notebook in both environments. 

When i evaluate the different losses i get highly divergent results.

I also checked the full .cfg with `cfg.dump()` - it is completely consistent.

I appreciate any help. Thanks in advance.

**1.Environment**

    Microsoft Azure - Machine Learning
    
    STANDARD_NC6
    
    Torch: 1.9.0+cu111

Results:

[![Results Azure][1]][1]

**2.Environment**

    GoogleColab free
    
    Torch: 1.9.0+cu111 

Results:

[![Results GoogleColab][2]][2]

**Detectron2 Parameters:**

    cfg = get_cfg()
    cfg.merge_from_file(model_zoo.get_config_file("COCO-Detection/retinanet_R_101_FPN_3x.yaml"))
    cfg.DATASETS.TRAIN = ("dataset_train",)
    cfg.DATASETS.TEST = ("dataset_test",)
    cfg.DATALOADER.NUM_WORKERS = 2
    cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url("COCO-Detection/retinanet_R_101_FPN_3x.yaml")  # Let training initialize from model zoo
    cfg.SOLVER.IMS_PER_BATCH = 2
    cfg.SOLVER.BASE_LR = 0.00025  # 0.00125 pick a good LR
    cfg.SOLVER.MAX_ITER = 1200    # 300 iterations seems good enough for this toy dataset; you will need to train longer for a practical dataset
    cfg.SOLVER.STEPS = []        # do not decay learning rate
    cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 512   # faster, and good enough for this toy dataset (default: 512)
    #cfg.MODEL.ROI_HEADS.NUM_CLASSES = 25  # only has one class (ballon). (see https://detectron2.readthedocs.io/tutorials/datasets.html#update-the-config-for-new-datasets)
    cfg.MODEL.RETINANET.NUM_CLASSES = 3
    # NOTE: this config means the number of classes, but a few popular unofficial tutorials incorrect uses num_classes+1 here.
    cfg.OUTPUT_DIR = "/content/drive/MyDrive/Colab_Notebooks/testrun/output"
    cfg.TEST.EVAL_PERIOD = 25

  [1]: https://i.stack.imgur.com/oKXID.png
  [2]: https://i.stack.imgur.com/uYQ6y.png

**Comment by Natriumpikant** [comment]
**Seems like the dataset was responsible for this issue - i cant reproduce the issue with a larger dataset, so everything seems to be fine.**
